src/windows/minidump_writer.rs

```diff
@@ -0,0 +1,279 @@
+    /// MiniDumpWriteDump that the pointers come from an external process so that
+    /// it can use eg ReadProcessMemory to get the contextual information from
+    /// the crash, rather than from the current process
+    is_external_process: bool,
+}
+
+impl MinidumpWriter {
+    /// Creates a minidump writer for a crash that occurred in an external process.
+    pub fn external_process(
+        crash_context: crash_context::CrashContext,
+        pid: u32,
+    ) -> Result<Self, Error> {
+        let crashing_process = unsafe {
+            OpenProcess(
+                GENERIC_ALL, // desired access
+                0,           // inherit handles
+                pid,         // pid
+            )
+        };
+
+        if crashing_process == 0 {
+            Err(std::io::Error::last_os_error().into())
+        } else {
+            Ok(Self {
+                crash_context,
+                crashing_process,
+                crashing_pid: pid,
+                is_external_process: true,
+            })
+        }
+    }
+
+    /// Creates a minidump writer for a crash that occurred in the current process.
+    ///
+    /// # Errors
+    ///
+    /// Fails if we are unable to open a `HANDLE` to the current process
+    pub fn current_process(crash_context: crash_context::CrashContext) -> Result<Self, Error> {
+        let crashing_pid = std::process::id();
+
+        // SAFETY: syscall
+        let crashing_process = unsafe {
+            OpenProcess(
+                GENERIC_ALL, // desired access
+                0,           // inherit handles
+                crashing_pid,
+            )
+        };
+
+        if crashing_process == 0 {
+            Err(std::io::Error::last_os_error().into())
+        } else {
+            Ok(Self {
+                crash_context,
+                crashing_process,
+                crashing_pid,
+                is_external_process: false,
+            })
+        }
+    }
+
+    /// Writes a minidump to the specified file
+    pub fn dump(&self, destination: &mut std::fs::File) -> Result<(), Error> {
+        let exc_info = if !self.crash_context.exception_pointers.is_null() {
+            // https://docs.microsoft.com/en-us/windows/win32/api/minidumpapiset/ns-minidumpapiset-minidump_exception_information
+            Some(md::MINIDUMP_EXCEPTION_INFORMATION {
+                ThreadId: self.crash_context.thread_id,
+                // This is a mut pointer for some reason...I don't _think_ it is
+                // actually mut in practice...?
+                ExceptionPointers: self.crash_context.exception_pointers as *mut _,
+                ClientPointers: if self.is_external_process { 1 } else { 0 },
+            })
+        } else {
+            None
+        };
+
+        // This is a bit dangerous if doing in-process dumping, but that's not
+        // (currently) a real target of this crate, so this allocation is fine
+        let mut user_streams = Vec::with_capacity(2);
```

Comment:
If this will interfere with in-process dumping, does in make sense to remove the `current_process()` function for now?

Reply:
This was just an offhand comment that might seem more ominous than it is, I don't think there's any reason to not have current process dumping, I'll just add a comment on it that users should note in-process dumping is less reliable just in general.